worker/worker.go

```diff
@@ -137,32 +183,38 @@ func (w *worker) pollPlayers(ctx context.Context) {
 				return nil
 			})
 			if err != nil {
-				w.l.Error("poll-session", "error", err)
+				w.l.Error("poll-players", "error", err)
 			}
 		}
 	}
 }
 
-var (
-	alliedTeams = []api.PlayerTeam{
-		api.PlayerTeamB8a,
-		api.PlayerTeamDak,
-		api.PlayerTeamGb,
-		api.PlayerTeamRus,
-		api.PlayerTeamUs,
-	}
-)
-
 func (w *worker) checkPlayer(ctx context.Context, p api.GetPlayerResponse) {
+	// If player is not spawned, reset first coordinate tracking and return
 	if !p.Position.IsSpawned() {
+		w.firstCoordSkipped.Delete(p.Id)
+		w.outsidePlayers.Delete(p.Id)
 		return
 	}
+
+	// Check if first coordinate has been skipped
+	if _, ok := w.firstCoordSkipped.Load(p.Id); !ok {
+		w.firstCoordSkipped.Store(p.Id, true)
+		return
+	}
+
+	// Calculate player's grid position
 	g := p.Position.Grid(w.current)
+
+	// Determine applicable fences based on team
 	var fences []data.Fence
 	if slices.Contains(alliedTeams, p.Team) {
 		fences = w.alliesFences
-	} else {
+	} else if slices.Contains(axisTeams, p.Team) {
 		fences = w.axisFences
+	} else {
```

Comment:
no need for the explicit else block here. Passing an uninitialized slice (like `fences`) to `len()` will always return 0. Hence, this would be covered with the next if condition.